consensus/istanbul/backend/engine.go

```diff
@@ -583,6 +628,12 @@ func (sb *Backend) StartValidating(hasBadBlock func(common.Hash) bool,
 	sb.processBlock = processBlock
 	sb.validateState = validateState
 
+	// Don't start core if we're a replica
+	blockNum := new(big.Int).Add(common.Big1, sb.currentBlock().Number())
```

Comment:
this function serves 2 purposes:

- enable (configure) core subsystems: Means that it should run, whenever we are primary
- start subsystem: run it

Going back to my idea of state machine, it would be nice to separate both concerns. Also, rn updateReplicaState ends up calling StartValidator for the `start` part, but not to configure it.

Reply:
Also, what about `Stop` (disable). For example if we have configured `startValidatingAt` but previously we call `stop`?

It's not really clear if this are orthogonal things (enable, versus primary/replica) or the same thing.

Reply:
I briefed tried splitting out the configure into `injectBlockProcessors` and then having a separate `StartValidating()`. It was less mixed, but I didn't feel much cleaner. Thinking about it more, I could split and then wrap `StartValidating` in miner with a call to `IsPrimaryForSeq`